### PR TITLE
Filter button and count for hidden tasks (#2156)

### PR DESCRIPTION
**Pull request by apopov77**
New button is added to task table's toolbar for filters menu.
A counter of hidden tasks appears when an filter is applied

### CODE REVIEW

File: ganttproject/src/main/java/biz/ganttproject/ganttview/TaskTable.kt [hunk 1]
```diff
@@ -101,6 +102,17 @@ class TaskTable(
   val taskTableModel = TaskTableModel(taskManager, taskManager.customPropertyManager)
   private val task2treeItem = mutableMapOf<Task, TreeItem<Task>>()
 
+  lateinit var hiddenTaskCountObserver : (Int) -> Unit
+  private var hiddenTaskCount : Int by Delegates.observable(0) {
```
Comment (dbarashev) [inline ganttproject/src/main/java/biz/ganttproject/ganttview/TaskTable.kt:106]: I suggest moving this counter into TaskFilterManager, and using SimpleIntegerProperty from JavaFX which allows for adding change listeners in the runtime:

```
class TaskFilterManager {
   ...
   val hiddenTaskCount = SimpleIntegerProperty(0)
   ...
 }
 ```
 
 ```
class TaskTable {
   ...
   this.filters.hiddenTaskCount.set(filteredCount)
  ...
}

class GanttChartTabContentPanel {
  ...
  this.filterManager.hiddenTaskCount.addListener(..)
  ...
}
```

Also, we probably can replace the call to TaskTable::taskFilterActions with the direct access of TaskFilterManager properties.

File: ganttproject/src/main/java/net/sourceforge/ganttproject/GanttChartTabContentPanel.java [hunk 2]
```diff
@@ -96,30 +102,41 @@ JComponent getComponent() {
...
-    var tableMenuButton = ToolbarKt.createButton(new TableMenuAction(), true);
-    Objects.requireNonNull(tableMenuButton).setOnAction(event -> {
+
+    Button tableFilterButton = ToolbarKt.createButton(new TableButtonAction("taskTable.tableMenuFilter"), true);
+    Objects.requireNonNull(tableFilterButton).setOnAction(event -> {
       tableMenu.getItems().clear();
-      myTaskTableSupplier.get().tableMenuActions(new MenuBuilderFx(tableMenu));
-      tableMenu.show(tableMenuButton, Side.BOTTOM, 0.0, 0.0);
+      myTaskTableSupplier.get().tableFilterActions(new MenuBuilderFx(tableMenu));
+      tableMenu.show(tableFilterButton, Side.BOTTOM, 0.0, 0.0);
```
Comment (dbarashev) [inline ganttproject/src/main/java/net/sourceforge/ganttproject/GanttChartTabContentPanel.java:110]: Rename `tableMenu` to `tableFilterMenu`?

File: ganttproject/src/main/java/biz/ganttproject/ganttview/TaskTable.kt [hunk 3]
```diff
@@ -101,6 +102,17 @@ class TaskTable(
   val taskTableModel = TaskTableModel(taskManager, taskManager.customPropertyManager)
   private val task2treeItem = mutableMapOf<Task, TreeItem<Task>>()
 
+  lateinit var hiddenTaskCountObserver : (Int) -> Unit
```
[the remark this answers is not in the document]
Comment (apopov77) [reply ganttproject/src/main/java/biz/ganttproject/ganttview/TaskTable.kt]: Removed

File: ganttproject/src/main/java/biz/ganttproject/ganttview/TaskTable.kt (hunk 1 above)
Comment (apopov77) [reply ganttproject/src/main/java/biz/ganttproject/ganttview/TaskTable.kt]: Done
Filter actions have been moved to filter manager

**Review by dbarashev — APPROVED** [review]
LGTM

**Pull request merged**